Lesson10/src/homework10/Buyer.java

```diff
@@ -0,0 +1,55 @@
+package homework10;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+
+public class Buyer extends Thread {
+    private List<CashRegister> cashRegisters;
+    private String currency;
+
+    public Buyer(String name, List<CashRegister> cashRegisters, String currency) {
+        super(name);
+        this.cashRegisters = cashRegisters;
+        this.currency = currency;
+    }
+
+    @Override
+    public void run() {
+        choiceCashRegister().serveCustomer(this);
+    }
+
+    private CashRegister choiceCashRegister() {
+        Random random = new Random();
+        List<CashRegister> similarCashRegister = new LinkedList<>();
+        for (CashRegister i : cashRegisters) {
+            if (i.getCurrency().equals(currency)) {
+                similarCashRegister.add(i);
+            }
+        }
+        if (similarCashRegister.size() == 0) {
+            System.out.println(getName() + "\tWe don't use this currency,  You could use our ATM");
+            return null;
```

Comment:
since you are calling choiceCashRegister().serveCustomer(this); in the run method - returning null will cause an exception. So you need to handle this situation in run method